hw/ip_templates/otp_ctrl/rtl/otp_ctrl.sv.tpl

```diff
@@ -530,15 +529,26 @@ module otp_ctrl
     hw2reg.direct_access_rdata = dai_rdata;
     // ANDing this state with dai_idle write-protects all DAI regs during pending operations.
     hw2reg.direct_access_regwen.d = direct_access_regwen_q & dai_idle;
-    // Assign these to the status register.
-    hw2reg.status = {part_errors_reduced,
-                     chk_timeout,
-                     lfsr_fsm_err,
-                     scrmbl_fsm_err,
-                     part_fsm_err[KdiIdx],
-                     fatal_bus_integ_error_q,
-                     dai_idle,
-                     chk_pending};
+    // Assign these to the status register. Note that the lower 2 bits of part_errors_reduced
+    // contain the DAI/LCI error. They are treated as normal erorrs and not part of the dedicated
+    // partitition status register.
+    hw2reg.status.partition_error.d      = |part_errors_reduced[$bits(part_errors_reduced)-1:2];
+    hw2reg.status.dai_error.d            = part_errors_reduced[0];
+    hw2reg.status.lci_error.d            = part_errors_reduced[1];
```

Comment:
I don't think this is correct, as DAI and LCI are at higher indices than 0 and 1, per `otp_ctrl_part_pkg` -- or am I missing sth?

Reply:
Good catch. I made it more robust by using the `PartitionIdx` definitions to lookup the status code from the partition status code vector. This is done for the DAI/LCI partitions but also for the generic partitions below.